sherpa/stats/tests/test_stats.py

```diff
@@ -240,388 +242,484 @@ def chi_nobkg(self, data, model, *args, **kwargs):
-                             result)
-
-    # bug #147
-    def test_wstat_calc_stat_info(self):
-        ui.load_pha("stat", self.make_path("3c273.pi"))
-        ui.set_source("stat", ui.powlaw1d.p1)
-        ui.set_stat("wstat")
-        ui.fit("stat")
-        ui.get_stat_info()
+            [1.750204250228727, 5.47466040324842, -1.9983562007031974]
+        )
+    }
+
+    compare_results(_fit_wstat_results_bench, results, tol=2e-4)
+
+
+# The following test passes if run by itself but fails when run with others
+# def test_wstat1(self):
+#     pha_fname = self.make_path("stats/acisf09122_000N001_r0013_pha3.fits")
+#     ui.load_pha(pha_fname)
+#     #ui.set_analysis('energy')
+#     ui.ignore(None, None)
+#     ui.notice(1.0, 1.6)
+#     src = ui.xsphabs.gal * ui.xspowerlaw.pl
+#     gal.nh = 0.1
+#     pl.phoindex = 0.7
+#     pl.norm = 1e-4
+#     ui.set_source(src)
+#     ui.set_stat('wstat')
+#     assert numpy.allclose(46.455049531, ui.calc_stat(), 1.e-7, 1.e-7)
+
+
+@requires_fits
+@requires_xspec
+@requires_data
+def test_wstat_error(hide_logging, reset_xspec, setup_bkg):
+    fit = Fit(setup_bkg['bkg'], setup_bkg['model'], WStat(), NelderMead())
+
+    with pytest.raises(StatErr):
+        fit.fit()
+
+
+def test_chi2datavar(hide_logging):
+    num = 3
+    xy = numpy.array(range(num))
+    ui.load_arrays(1, xy, xy, Data1D)
+    ui.set_stat('chi2datavar')
+    err = ui.get_staterror()
+    assert err == pytest.approx(numpy.sqrt(xy))
+
+
+@requires_fits
+@requires_xspec
+@requires_data
+def test_get_stat_info(hide_logging, make_data_path):
+    fname_3c273 = make_data_path("3c273.pi")
+    ui.load_pha(fname_3c273)
+    src = ui.xspowerlaw.pl
+    ui.set_source(src)
+    ui.guess('pl')
+    ui.set_stat('wstat')
+    stat_info = ui.get_stat_info()[0]
+    assert stat_info.dof == 44
+    assert stat_info.numpoints == 46
+
+
+@requires_fits
+@requires_xspec
+@requires_data
+@pytest.mark.parametrize('stat', [MySimulStat, MyChiNoBkg])
+def test_simul_stat_fit(stat, hide_logging, reset_xspec, setup_two):
+    data1 = setup_two['data_pi2278']
+    data2 = setup_two['data_pi2286']
+    model1 = setup_two['model_pi2278']
+    model2 = setup_two['model_pi2286']
+    data = DataSimulFit(name='data1data2', datasets=[data1, data2])
+    model = SimulFitModel(name='model1model2', parts=[model1, model2])
+    fit = Fit(data=data, model=model, stat=stat(),
+              method=NelderMead())
+    result = fit.fit()
+
+    _fit_simul_datavarstat_results_bench = {
+        'succeeded': 1,
+        'numpoints': 18,
+        'dof': 15,
+        'istatval': 56609.70689926489,
+        'statval': 126.1509268988255,
+        'parvals': numpy.array(
+            [0.8417576197443695, 1.6496933246579941, 0.2383939869443424]
+        )
+    }
+
+    compare_results(_fit_simul_datavarstat_results_bench,
+                    result)
+
+
+@requires_data
+@requires_fits
+def test_wstat_calc_stat_info(hide_logging, make_data_path, clean_astro_ui):
```

Comment:
Just to note that in working on #1060 I came across a run (using `pytest -n6` to run in parallel` where this test (that is, the old version, it this PR) failed, because `clean_astro_ui` - or equivalent - had not been used. That is, there are reasons for these changes!